### PR TITLE
Reindex Trellis into ElasticSearch

### LINKED ISSUE #15: Index RDF into a separate ES Index
Depending on the ActiveMQ message, index payload into one of two target ES indices, one for Non-RDF JSON resource templates  and one for RDF JSON-LD.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes LD4P/sinopia_server#45

Includes:

* Add Crawler and Reindexer classes to handle bulk of reindexing work
* Add ability to Indexer to recreate indices when reindexing
* Add `bin/reindex` task and wrap a new `docker-compose` service around it for testing
* Extract Waiter class from runner so that it can be used in runner and reindexing work, and add tests
* Move type-checking functionality into the Request class
* Remove `body()` from Request and add `response()` to expose both response bodies and headers
* Use consistent capitalization, matching filename to class name
* Name files (mocks in particular) consistently
* Add optional chaining proposal via Babel (safe navigation/"Elvis" operator)
* Prefer `it()` to `test()` when test descriptions are verb-y (like RSpec)
* Use `test.todo()` for tests we haven't figured out how to write yet
* Collect coverage from all JS files under `src/`

### CODE REVIEW

File: src/Config.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 export default class Config {
   static get platformUrl() {
-    return process.env.TRELLIS_BASE_URL || 'http://platform:8080'
+    // If configured in environment (hence, prod-like), use that value
```
Comment (jermnelson) [inline src/Config.js]: I'm still not  entirely convinced about the advantages having a class for `Config` is necessary for functional javascript

File: src/Config.js [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+    if (process.env.TRELLIS_BASE_URL)
+      return process.env.TRELLIS_BASE_URL
+
+    // Else, we're in a dev/test env, so use localhost if not in container, or
+    // container value if in container
+    return Boolean(process.env.INSIDE_CONTAINER) ? 'http://platform:8080' : 'http://localhost:8080'
```
Comment (jermnelson) [inline src/Config.js]: I like the use of Boolean here for testing all of the different ways this environmental variable can be false

File: src/Config.js (hunk 1 above)
Comment (mjgiarlo) [reply src/Config.js]: @jermnelson I'd be happy to put in a follow-up PR that dumps the bespoke `Config` class in favor of the node `config` gem. It seems to me that our projects have already been moving in that direction. Is that cool by you?

Comment (mjgiarlo) [reply src/Config.js]: Does this suggest too that https://github.com/LD4P/sinopia_editor/issues/457 should be in scope? These are our last two codebases, AFAIK, that use a custom `Config` class.

Comment (jermnelson) [reply src/Config.js]: The bespoke `Config` class here is fine for now; it works and is tested. I don't think it is critical path but maybe better noted as a future fix if you want to make a ticket.

Comment (mjgiarlo) [reply src/Config.js]: @jermnelson OK! I'll write it up and tag it as "future"

**Pull request merged**